src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -80,6 +80,7 @@ export class TerminalService implements ITerminalService {
 			this.setActiveInstanceByIndex(0);
 		}
 		this._onInstancesChanged.fire();
+		terminalInstance.setCommandsToSkipShell(this.configHelper.getCommandsToSkipShell());
```

Comment:
Ok, after taking a closer look, there is quite some refactoring involved to do this right. _updateCursorBlink, _updateCommandsToSkipShell and _updateScrollback have to move too, makes no sense to leave them in the panel. But _updateFont has to move out of the old config update and stay on its own,  its basically the same "class" of responsibility as _updateTheme. That means the event listeners need to be modified too… I'll figure it out, will take a little while until complete…